client/express-checkout/index.js

```diff
@@ -177,50 +188,146 @@ jQuery( ( $ ) => {
 		 * @param {Object} options ECE options.
 		 */
 		startExpressCheckoutElement: ( options ) => {
+			const shippingRates = options.displayItems
+				.filter( ( i ) => i.label === 'Shipping' )
+				.map( ( i ) => ( {
+					id: `rate-${ i.label }`,
+					amount: i.amount,
+					displayName: i.label,
+				} ) );
+
+			// This is a bit of a hack, but we need some way to get the shipping information before rendering the button, and
+			// since we don't have any address information at this point it seems best to rely on what came with the cart response.
+			// Relying on what's provided in the cart response seems safest since it should always include a valid shipping
+			// rate if one is required and available.
+			// If no shipping rate is found we can't render the button so we just exit.
+			if ( options.requestShipping && ! shippingRates ) {
+				return;
+			}
+
 			const elements = api.getStripe().elements( {
 				mode: options?.mode ?? 'payment',
 				amount: options?.total,
 				currency: options?.currency,
+				paymentMethodCreation: 'manual',
 			} );
 
 			const eceButton = wcpayECE.createButton( elements, {
 				buttonType: {
-					googlePay: wcpayExpressCheckoutParams.button.type,
-					applePay: wcpayExpressCheckoutParams.button.type,
+					googlePay: getExpressCheckoutData( 'button' ).type,
+					applePay: getExpressCheckoutData( 'button' ).type,
 				},
 			} );
 
 			wcpayECE.showButton( eceButton );
 
-			wcpayECE.attachButtonEventListeners( eceButton );
-
 			eceButton.on( 'click', function ( event ) {
 				const clickOptions = {
-					business: {
-						name: 'Mikes Bikes',
-					},
-					lineItems: [
-						{ name: 'Bike', amount: 200 },
-						{ name: 'Helmet', amount: 300 },
-					],
-					shippingAddressRequired: true,
-					shippingRates: [
-						{
-							id: '1',
-							amount: 500,
-							displayName: 'Standard Shipping',
-						},
-						{
-							id: '2',
-							amount: 1000,
-							displayName: 'Expedited Shipping',
-						},
-					],
+					lineItems: options.displayItems.map( ( i ) => ( {
+						...i,
+						name: i.label,
+					} ) ),
+					emailRequired: true,
+					shippingAddressRequired: options.requestShipping,
+					phoneNumberRequired: options.requestPhone,
+					shippingRates,
 				};
+				wcpayECE.block();
 				event.resolve( clickOptions );
 			} );
 
+			// FIXME: This handler is copied from ./event-handlers.js. We should re-use the same function here.
+			eceButton.on( 'shippingaddresschange', async ( event ) => {
```

Comment:
@reykjalin I think we are duplicating efforts. Check my PR https://github.com/Automattic/woocommerce-payments/pull/8914/files#diff-71dc9212813424406f6e197c711b7b1776d744686daba30456192cfdadbe99b2R224-R226

Reply:
I'm re-using the handlers defined in `event-handlers.js`. I thought it was meant to re-use it? @rafaelzaleski

Reply:
Oh we probably are! Although I'm not sure passing the elements in works since the blocks implementation doesn't seem to have access to those 🤔

https://github.com/Automattic/woocommerce-payments/blob/95b74a487a7811604097acfd67dac428bfc0e149/client/express-checkout/blocks/components/express-checkout-component.js#L15-L50

I think the blocks implementation needs to update the `buttonOptions`? Maybe? Oh or maybe just expose the elements in the `useExpressCheckout` hook I guess 🤔 

I'll work it into my PR so I can incorporate your changes as well 👍

Reply:
Maybe we should use the same branch?

Reply:
I added my take on this in 0973d1f2193281f5d3288bb319720d3728742937

Reply:
Merged this PR into yours